string.c

```diff
@@ -3110,7 +3110,7 @@ rb_str_resize(VALUE str, long len)
     int independent = str_independent(str);
     long slen = RSTRING_LEN(str);
 
-    if (slen > len && ENC_CODERANGE(str) != ENC_CODERANGE_7BIT) {
+    if (slen != len && ENC_CODERANGE(str) != ENC_CODERANGE_7BIT) {
```

Comment:
With moving `termlen` from the following block to here, and:
```suggestion
    if ((slen > len && ENC_CODERANGE(str) != ENC_CODERANGE_7BIT) ||
        (termlen > 1 && (slen % termlen || len % termlen))) {
```

Reply:
Changed it to `(slen % termlen == 0) != (len % termlen == 0)` because I think it's more precise